juriscraper/opinions/united_states/state/miss.py

```diff
@@ -1,100 +1,85 @@
 
-    """Keep track of the most recent N date pages.
-    We dont want to crawl all the way back to 1996, so we only
-    parse the most recent [self.number_of_dates_to_process]
-    number of date pages.  Since cases are usually published
-    once a week, this means scraping about the most recent
-    months worth of cases.
-    """
+    def make_dates(self) -> None:
+        """Get last five dates sort and download them
 
-    def parse_date_pages(self, dates_page):
-        # For testing, each example file should be a specific sub-date page,
-        # like https://courts.ms.gov/Images/HDList/SCT02-27-2020.html
-        if self.test_mode_enabled():
-            # date below is arbitrary and doesnt matter, it just
-            # needs to be static for testing to work
-            self.pages["2020-02-28"] = dates_page
-            return
-        for date in self.get_dates_from_date_page(dates_page):
-            url = "{}/Images/HDList/SCT{}.html".format(
-                self.domain,
-                datetime.date.strftime(date, "%m-%d-%Y"),
-            )
-            page = self._get_html_tree_by_url(url)
-            self.pages[f"{date}"] = page
+        :param html:The HTML for the initial call to the site
+        :return:None
+        """
+        dates = json.loads(self.html.xpath("//p/text()")[0])
+        dates.sort(
+            key=lambda date: datetime.strptime(date, "%m-%d-%Y"), reverse=True
+        )
+        self.dates = dates[:5]
 
-    """Convert string of dates on page into list of date objects.
-    """
+    def _process_html(self) -> None:
+        """Process the HTML for each calendar date.
 
-    def get_dates_from_date_page(self, dates_page):
-        dates = []
-        substrings = dates_page.text_content().split('"')
-        for substring in substrings:
-            try:
-                dates.append(convert_date_string(substring))
-            except ValueError:
-                pass
-        dates.sort(reverse=True)
-        return dates[: self.number_of_dates_to_process]
+        :return:None
+        """
 
-    def _process_html(self):
-        for date, page in self.pages.items():
-            for anchor in page.xpath(".//a[contains(./@href, '.pdf')]"):
-                parent = anchor.getparent()
+        # Find the last five dates
+        if not self.dates:
+            self.make_dates()
 
-                # sometimes the first opinion on the pages is nested
-                # in a <p> tag for whatever reason.
-                while parent.getparent().tag != "body":
-                    parent = parent.getparent()
+        # Iterate over the dates to generate the HTML to parse
+        for date in self.dates:
+            if not self.test_mode_enabled():
+                self.date = date
+                self.parameters["date"] = self.date
+                self.url = "https://courts.ms.gov/appellatecourts/docket/get_hd_file.php"
+                self.html = super()._download()
```

Comment:
This still needs a DeferringList, but that's all I see the remains.

Reply:
Great.  Thanks.